src/core/execute-serialize.c

```diff
@@ -1218,7 +1250,8 @@ static int exec_runtime_deserialize(ExecRuntime *rt, FILE *f, FDSet *fds) {
                         r = free_and_strdup(&rt->shared->var_tmp_dir, val);
                         if (r < 0)
                                 return r;
-                } else if ((val = startswith(l, "exec-runtime-netns-storage-socket="))) {
+                } else if ((val = startswith(l, "exec-runtime-netns-storage-socket=")) ||
+                                (val = startswith(l, "exec-runtime-netns-storage-socket-by-fd-index="))) {
```

Comment:
this is a big ugly, i'd prefer if we'd leave the field names as they are, but change what we serialize, i.e. prefix with a char like `#` if they are indexes. that way the index-vs-fd thing can be hidden within a fnction thhat serializes individual fds, and doesn't leak all over the place into all code that might want to serialize either.